src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs

```diff
@@ -151,23 +151,20 @@ public override object ConvertTo(ITypeDescriptorContext context, CultureInfo cul
 
                 if (value == null || value == CultureInfo.InvariantCulture)
                 {
-                    retVal = defaultCultureString;
+                    return defaultCultureString;
                 }
-                else
+                if (value is CultureInfo c)
                 {
-                    retVal = GetCultureName(((CultureInfo)value));
+                    return GetCultureName(c);
                 }
-
-                return retVal;
             }
             if (destinationType == typeof(InstanceDescriptor) && value is CultureInfo)
```

Comment:
>value is CultureInfo [](start = 65, length = 20)

nit: may be you can make this as

value is CultureInfo c

and then remove the line 

CultureInfo c = (CultureInfo)value;

Reply:
Sadly we can't do this, as we use it in the check for `destinationType == typeof(string)` - therefore we'd have to nest both the `destinationType` check statements to avoid build errors. However, this would be breaking, as it changes the behavior for `ConvertTo(null, typeof(string))`

Reply:
Is this true even using different name? something like 

value is CultureInfo culture?

---
In reply to: [299625897](https://github.com/dotnet/corefx/pull/38980#discussion_r299625897) [](ancestors = 299625897)